### ISSUE TITLE
Error in Log: FilesUploadWorker: Error uploading

### ISSUE THREAD

**Issue by TheColin21** [nextcloud/android#11475]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

- have a phone with some old and some new raw images in the same folder
- setup auto upload for this folder (including upload of existing images)
- after some time, notice that only the new ones get uploaded
- run adb logcat, notice the bug

### Expected behaviour

all files get uploaded

### Actual behaviour

only new files get uploaded, the old ones throw a log message like the one below

### Android version

13

### Device brand and model

Pixel 6 Pro

### Stock or custom OS?

Stock

### Nextcloud android app version

3.24.1

### Nextcloud server version

25.0.5

### Using a reverse proxy?

Yes

### Android logs

```
03-24 18:30:06.469 14992 15054 E FilesUploadWorker: Error uploading
03-24 18:30:06.469 14992 15054 E FilesUploadWorker: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.owncloud.android.datamodel.OCFile.getDecryptedRemotePath()' on a null object reference
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at com.owncloud.android.utils.FileStorageUtils.checkEncryptionStatus(FileStorageUtils.java:469)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at com.owncloud.android.operations.UploadFileOperation.run(UploadFileOperation.java:435)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:205)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at com.nextcloud.client.jobs.FilesUploadWorker.upload(FilesUploadWorker.kt:158)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at com.nextcloud.client.jobs.FilesUploadWorker.handlePendingUploads(FilesUploadWorker.kt:106)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at com.nextcloud.client.jobs.FilesUploadWorker.doWork(FilesUploadWorker.kt:89)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at androidx.work.Worker$1.run(Worker.java:86)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
03-24 18:30:06.469 14992 15054 E FilesUploadWorker:     at java.lang.Thread.run(Thread.java:1012)
03-24 18:30:06.469 14992 15054 D UploadsStorageManager: updateDatabaseUploadResult uploadResult: RemoteOperationResult{mSuccess=false, mHttpCode=-1, mHttpPhrase='null', mException=java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.owncloud.android.datamodel.OCFile.getDecryptedRemotePath()' on a null object reference, mCode=HOST_NOT_AVAILABLE, message='null', getLogMessage='Unexpected exception'} upload: com.owncloud.android.operations.UploadFileOperation@9077328
03-24 18:30:06.478 14992 15054 V UploadsStorageManager: Updating /storage/emulated/0/Pictures/Raw/IMG_20210923_035146625.dng with status:UPLOAD_FAILED and result:NETWORK_CONNECTION (old:/storage/emulated/0/Pictures/Raw/IMG_20210923_035146625.dng status:UPLOAD_IN_PROGRESS result:-1)
03-24 18:30:06.478 14992 15054 V UploadsStorageManager: Updating /storage/emulated/0/Pictures/Raw/IMG_20210923_035146625.dng with status=UPLOAD_FAILED
03-24 18:30:06.493 14992 15054 D UploadsStorageManager: updateUpload returns with: 1 for file: /storage/emulated/0/Pictures/Raw/IMG_20210923_035146625.dng
03-24 18:30:06.493 14992 15054 D UploadsStorageManager: notifyObserversNow
03-24 18:30:06.497 14992 15054 D UploadsStorageManager: Retrieve job com.owncloud.android.db.OCUpload@6326841 for id 3132
03-24 18:30:06.497 14992 15054 V UploadsStorageManager: Updating /storage/emulated/0/Pictures/Raw/IMG_20210327_123214.dng with status=UPLOAD_IN_PROGRESS
03-24 18:30:06.503 14992 15004 W System  : A resource failed to call AbstractCursor.close.
03-24 18:30:06.503 14992 15004 W System  : A resource failed to call CursorWrapperInner.close.
03-24 18:30:06.513 14992 15004 W System  : A resource failed to call AbstractCursor.close.
03-24 18:30:06.514 14992 15004 W System  : A resource failed to call CursorWrapperInner.close.
03-24 18:30:06.516 14992 15004 W System  : A resource failed to call AbstractCursor.close.
03-24 18:30:06.516 14992 15004 W System  : A resource failed to call CursorWrapperInner.close.
03-24 18:30:06.517 14992 15054 D UploadsStorageManager: updateUpload returns with: 1 for file: /storage/emulated/0/Pictures/Raw/IMG_20210327_123214.dng
03-24 18:30:06.517 14992 15054 D UploadsStorageManager: notifyObserversNow
03-24 18:30:06.518 14992 15004 W System  : A resource failed to call AbstractCursor.close.
03-24 18:30:06.518 14992 15004 W System  : A resource failed to call CursorWrapperInner.close.
03-24 18:30:06.553 14992 15004 W System  : A resource failed to call AbstractCursor.close.
03-24 18:30:06.553 14992 15004 W System  : A resource failed to call CursorWrapperInner.close.

### Server error logs

```bash
no errors on the server side
```

### Additional information

device is rooted with Magisk 25.2. This should not have anything to do with this bug though. I am running the current version (March 2023) of the stock Pixel ROM and there are no Magisk modules or anything similar that even remotely have something to do with nextcloud.

**Comment by joshtrichards** [comment]
Are you using E2E?

**Comment by TheColin21** [comment]
No

Am 30. März 2023 23:36:15 MESZ schrieb Josh Richards ***@***.***>:

**Comment by joshtrichards** [comment]
Dup of #11316 and there's a fix merged into the v3.24 tree. 

It's not in v3.24.1 however.
